python/tskit/trees.py

```diff
@@ -6351,3 +6352,79 @@ def newick_trees(self, precision=3, breakpoints=None, Ne=1):
             "This method is no longer supported. Please use the Tree.newick"
             " method instead"
         )
+
+
+def write_ms(
+    tree_sequence,
+    output,
+    print_trees=False,
+    precision=4,
+    mutation_rate=0,
+    num_replicates=0,
+):
+    """
+    Write ``ms`` formatted output from the genotypes of a tree sequence
+    or an iterator over tree sequences. Usage:
+
+    .. code-block:: python
+
+        import tskit as ts
+
+        tree_sequence = msprime.simulate(
+            sample_size=sample_size,
+            Ne=Ne,
+            length=length,
+            mutation_rate=mutation_rate,
+            recombination_rate=recombination_rate,
+            random_seed=random_seed,
+            num_replicates=num_replicates,
+        )
+        with open("output.ms", "w") as ms_file:
+            ts.write_ms(tree_sequence, ms_file, mutation_rate=mutation_rate)
+
+    :param io.IOBase output: The file-like object to write the ms-style output
+    :param bool print_trees: Boolean parameter to write out newick format trees
+        to output [optional]
+    :param int precision: Numerical precision with which to write the ms
+        output [optional]
+    :param float mutation_rate: Mutation rate used in the simulation
+    :param int num_replicates: Number of replicates simulated [required if
+        num_replicates used in simulation]
+
+    The first line of this ms-style output file written has two arguments which
+    are sample size and number of replicates. The second line has a 999 as a substitute
+    for the random seed.
+    """
+    if isinstance(tree_sequence, collections.Iterable):
+        i = 0
+        for tree_seq in tree_sequence:
+            if i == 0:
+                writer = ms.msWriter(
+                    tree_seq,
+                    print_trees=print_trees,
+                    precision=precision,
+                    mutation_rate=mutation_rate,
+                    num_replicates=num_replicates,
+                    write_header=True,
+                )
+            else:
+                writer = ms.msWriter(
+                    tree_seq,
+                    print_trees=print_trees,
+                    precision=precision,
+                    mutation_rate=mutation_rate,
+                    num_replicates=num_replicates,
+                    write_header=False,
+                )
```

Comment:
Here you can have one call to msWriter with `write_header=(i==0)`

Reply:
Implemented this change.